feconf.py

```diff
@@ -61,8 +61,23 @@
     FRONTEND_TEMPLATES_DIR = os.path.join('core', 'templates', 'dev', 'head')
 DEPENDENCIES_TEMPLATES_DIR = (
     os.path.join(EXTENSIONS_DIR_PREFIX, 'extensions', 'dependencies'))
+
+if not constants.DEV_MODE:
+    VALUE_GENERATORS_DIR_FOR_JS = os.path.join(
```

Comment:
Hm, I think this is a bit confusing. The name sort of suggests that the visualization code is actually split with the JS code in one place and non-JS code somewhere else.

How about having VALUE_GENERATORS_UNMINIFIED_JS_DIR and VALUE_GENERATORS_MINIFIED_JS_DIR be constants, and then select when you need them, based on the value of DEV_MODE?

Reply:
I am a bit confused. The name suggests that js code is in one place and non-js elsewhere which is right because we have js files only after compilation and in the value generators dir we have html templates and ts code.

The constant is used at three places:

1. [To obtain html templates](https://github.com/oppia/oppia/blob/develop/core/domain/value_generators_domain.py#L59)
2. [To obtain js template](https://github.com/oppia/oppia/blob/develop/core/domain/value_generators_domain.py#L74)
3. [To obtain models](https://github.com/oppia/oppia/blob/develop/core/domain/value_generators_domain.py#L104)

So, I feel it is fine to have js and non-js split. I am not sure if I understood your query correctly here, so please correct me on this if anything is wrong.

Reply:
I think the structure of the code is probably OK, but I was suggesting a rename, so that you clearly have unminified vs minified (if that's the distinction). I also think it's perfectly fine to migrate the directory fully (i.e. not just the JS files -- you can copy over the py/html/etc. files as well) so that we can just look for everything in one place depending on the mode. It's conceptually easier to do that, instead of having the JS files and the HTML/etc. files in separate places.

Reply:
The value_generators js files are obtained always from local_compiled_js irrespective of dev mode. So, we do not need two cases now.